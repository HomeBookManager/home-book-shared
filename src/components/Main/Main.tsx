import { BrowserRouter as Router } from 'react-router-dom';
import { FC } from 'react';

// components
import AppHelmet from '../AppHelmet/AppHelmet';

// core
import Routing, { TRoutingProps } from '../../core/Routing/Routing';

export type TMainProps = TRoutingProps;

export const Main: FC<TMainProps> = (props) => {
  return (
    <Router>
      <AppHelmet />
      <Routing {...props} />
    </Router>
  );
};

export default Main;
