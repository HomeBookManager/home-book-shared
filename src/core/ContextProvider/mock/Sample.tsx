import { useContext } from 'react';

// @home-book
import { Theme } from 'home-book-types';

// core
import { Context } from '../ContextProvider';

const Sample = () => {
  const { theme, setTheme } = useContext(Context);

  return (
    <div>
      {theme}
      <button onClick={() => setTheme(Theme.dark)}>Click</button>
    </div>
  );
};

export default Sample;
