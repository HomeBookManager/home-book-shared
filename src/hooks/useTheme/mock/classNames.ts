export const classNames = {
  className: 'className',
  nestedClassName: {
    modificators: { nestedClassName: 'nestedClassName--modificator' },
    name: 'nestedClassName',
  },
};

export const classNamesWithTheme = {
  className: 'className className--dark',
  nestedClassName: {
    modificators: {
      nestedClassName:
        'nestedClassName--modificator nestedClassName--modificator--dark',
    },
    name: 'nestedClassName nestedClassName--dark',
  },
};
