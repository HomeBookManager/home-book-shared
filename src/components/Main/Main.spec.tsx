import { HelmetProvider } from 'react-helmet-async';
import { render } from '@testing-library/react';

// components
import Main from './Main';

// mocks
import { appRoutesData, routes } from '../../tests/mock/routing';

describe('Main snapshots', () => {
  it('should render Main', async () => {
    // mock
    const notFoundPage = <>Not Found Page</>;

    // before
    const { asFragment } = render(
      <HelmetProvider>
        <Main
          appRoutesData={appRoutesData}
          notFoundPage={notFoundPage}
          routes={routes}
        />
      </HelmetProvider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
