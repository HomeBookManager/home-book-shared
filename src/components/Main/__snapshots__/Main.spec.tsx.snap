// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main snapshots should render Main 1`] = `
<DocumentFragment>
  Component
</DocumentFragment>
`;
